src/builders/enzymes.jl

```diff
@@ -113,3 +131,217 @@ function gene_product_isozyme_constraints(
 end
 
 export gene_product_isozyme_constraints
+
+"""
+$(TYPEDSIGNATURES)
+
+Returns freshly allocated variables `fluxes_reverse`, `fluxes_forward`,
+`isozyme_forward_amounts`, `isozyme_reverse_amounts`, and
+`gene_product_amounts`, which is used to build enzyme constrained models.
+"""
+function enzyme_variables(
+    constraints::C.ConstraintTree;
+    fluxes = constraints.fluxes,
+    reaction_isozymes::Dict{String,Dict{String,Isozyme}},
+)
+    gene_ids = unique([
+        Symbol(gid) for isos_dict in values(reaction_isozymes) for
+        iso in values(isos_dict) for gid in keys(iso.gene_product_stoichiometry)
+    ])
+
+    # might be nice to omit some conditionally (e.g. slash the direction if one
+    # kcat is nothing)
+    isozyme_amounts = isozyme_amount_variables(
+        Symbol.(keys(reaction_isozymes)),
+        rid -> Symbol.(keys(reaction_isozymes[string(rid)])),
+    )
+
+    # allocate variables for everything (nb. += wouldn't associate right here)
+    unidirectional_fluxes(
+        constraints;
+        fluxes,
+    ) +
+    :isozyme_forward_amounts^isozyme_amounts +
+    :isozyme_reverse_amounts^isozyme_amounts +
+    :gene_product_amounts^C.variables(keys = gene_ids, bounds = C.Between(0, Inf))
+end
+
+export enzyme_variables
+
+"""
+$(TYPEDSIGNATURES)
+
+Returns enzyme and protein capacity constraints, using the variables: `fluxes`,
+`fluxes_reverse`, `fluxes_forward`, `isozyme_forward_amounts`,
+`isozyme_reverse_amounts`, and `gene_product_amounts`. These variables should
+already be present in the model, see [`enzyme_variables`](@ref) for a quick way
+to add them to a normal flux balance model.
+
+Only reactions in `reaction_isozymes`, which is a mapping of reaction
+identifiers to [`Isozyme`](@ref) descriptions are included in the constraints.
+For each gene product in `reaction_isozymes`, a corresponding entry in
+`gene_product_molar_masses` must be present, which is a mapping of gene products
+to their molar masses.
+
+`capacity` (mass/gDW units) may be a single number, which sets the combined
+limit for all the gene products contained in `reaction_isozymes`. Alternatively,
+`capacity` may be a vector of identifier-genes-limit triples that make a
+constraint (identified by the given identifier) that limits the listed genes to
+the given limit (mass/gDW units).
+
+Note: [`simplified_enzyme_constraints`](@ref) and [`enzyme_constraints`](@ref)
+differ in how the capacity bound(s) are formulated. For the former, fluxes are
+used, but for the latter, gene products are used directly.
+"""
+function enzyme_constraints(
+    constraints;
+    fluxes = constraints.fluxes,
+    fluxes_reverse = constraints.fluxes_reverse,
+    fluxes_forward = constraints.fluxes_forward,
+    isozyme_forward_amounts = constraints.isozyme_forward_amounts,
+    isozyme_reverse_amounts = constraints.isozyme_reverse_amounts,
+    gene_product_amounts = constraints.gene_product_amounts,
+    reaction_isozymes::Dict{String,Dict{String,Isozyme}},
```

Comment:
no Isozyme possible here, I'll convert this to getter functions.

Reply:
Not a super fan of the getters, it just adds another layer for the user